### DISCUSSION TITLE
I couldnt find authorize url in development document

### DISCUSSION THREAD

**Discussion by gozenunten** [tidal-music/discussions#43] [Q&A]
I already read all document but I couldnt find any authorize url in document. Also I create my application from developer dashbaord but I cant set homePage and privacy policy link. Am I need some verification for that? How can I request for get authorize and access token? can someone explain this process? or maybe Tidal Development Team can add to document.

**Comment by stefanmoro** [comment]
Hey @gozenunten,

As of today, the only way to create a valid access token for querying the TIDAL API is by using the Client Credentials flow. You can find all relevant documentation [here](https://developer.tidal.com/documentation/authorization/authorization-overview).

We are currently working on, and plan to open up in the next few months, functionality for logging in TIDAL users via 3rd party apps. Initially we are targeting basic access to users playlists, but will expand throughout the year.

Regarding setting homepage and privacy policy link, that is something we aim to add to the dashboard going forward as well.

**Reply by gozenunten** [reply]
Thank your for reply @stefanmoro,

When I checked other similar apps I seen this url "https://login.tidal.com/authorize?lang=en&response_type=code&redirect_uri=\(redirectUri)&client_id=\(clientId)&code_challenge=\(codeChallenge)" but I didnt see this url in document. Is this url valid or deprecated? Or where am I request for authorize and access token?

note: Im developing iOS App with swift.

**Reply by stefanmoro** [reply]
Hey again, 

That endpoint is not yet available for open use, and any such use today represents an unauthorized use of our API.

This is top focus for us at the moment, so quite soon (within a few months) you will see similar functionality becoming available for TIDAL Developer Platform.

**Reply by gozenunten** [reply]
We are developing an iOS application with many different music platform. Also we want to implement Tidal Music. So if Tidal Music give early api access permission to us, we can add Tidal Music to our iOS application. How can we request as a company. Can you help us?

**Reply by appleseedexm** [reply]
Is there any plan or even roadmap when code flow will be made available? Is there beta access available? 

It seems like there's dire need to support some official way of accessing user specific resources. Currently the (OSS) community has no official way of writing software for Tidal users.

**Reply by stefanmoro** [reply]
Hey @appleseedexm,

The [auth code flow](https://developer.tidal.com/documentation/authorization/authorization-authorization-code) was actually made available a few months ago and you can now login users through your app.

Our top priority right now is to open up access to "user data" endpoints. We will open up new endpoints as soon as they are ready, and we hope that the first ones will be opened up within the next week or two.

You can see the list of [scopes](https://developer.tidal.com/documentation/authorization/authorization-scopes) for an indication of the functionality we are focusing on opening up.

**Reply by appleseedexm** [reply]
I had a deeper look at this and I dont understand the current state of the API, especially code flow. 
Code flow is available, but no scopes? That would render it useless, no? What should I login to if there's no open scopes? And the documentation does not mention that these scopes are "soon to come", but give the impression that they're available?

Any advice would be appreciated. In the meantime I opened a ticket with support since that's the only information I get from the API when trying to use code flow.

**Reply by stefanmoro** [reply]
Hey @appleseedexm ,

Please refer to our API reference for the single source of truth about what functionality is currently available, and which endpoints are protected by which scopes. [catalogue](https://developer.tidal.com/apiref?spec=catalogue-v2), [search](https://developer.tidal.com/apiref?spec=search-v2), [user](https://developer.tidal.com/apiref?spec=user-v2), [playlists](https://developer.tidal.com/apiref?spec=user-playlist-v2). 

I saw that a few scopes listed at [scopes](https://developer.tidal.com/documentation/authorization/authorization-scopes) documentation page are not in use just yet. I removed them from the docs until we open up endpoints protected by those scopes.

**Reply by appleseedexm** [reply]
Glad that you fixed the scopes. 

I still get the same error that I should contact customer support. Customer support sends me here, so dead end. 
Not really sure what to do, but with the read-only scopes it's no use to look into this anyway, maybe this will work once additional scopes get released. Am I right that there's currently no official way to write data for actual Tidal users?
